codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -984,6 +998,7 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
     private fun generateParsePercentEncodedStrAsStringFn(binding: HttpBindingDescriptor): RuntimeType {
         val output = symbolProvider.toSymbol(binding.member)
         val fnName = generateParseStrFnName(binding)
+        val type = output.extractSymbolFromOption()
```

Comment:
nit: this is technically a `Symbol`, not a `RustType`, so `symbol` would be a more appropriate variable name.